src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,159 @@
+            QLatin1String(" --stop-address=0x") +
+            QString::number(m_curSymbol.relAddr + m_curSymbol.size, 16) +
+            QLatin1String(" ") + m_curSymbol.path;
+
+    showDisassembly(processName);
+}
+
+/**
+ *  Produce disassembler with 'objdump' and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassembly(QString processName) {
+    QTemporaryFile m_tmpFile;
+    QProcess asmProcess;
+
+    if (m_tmpFile.open()) {
+        asmProcess.start(processName);
+
+        if (!asmProcess.waitForStarted() || !asmProcess.waitForFinished()) {
+            return;
+        }
+        QTextStream stream(&m_tmpFile);
+        stream << asmProcess.readAllStandardOutput();
+        m_tmpFile.close();
+    }
+
+    if (m_tmpFile.open()) {
+        int row = 0;
+        QStandardItemModel *model = new QStandardItemModel();
+
+        QStringList headerList;
+        headerList.append(QLatin1String("Assembly"));
+        model->setHorizontalHeaderLabels(headerList);
+
+        QTextStream stream(&m_tmpFile);
+        while (!stream.atEnd()) {
+            QString asmLine = stream.readLine();
+            if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly"))) continue;
+
+            QStandardItem *asmItem = new QStandardItem();
+            asmItem->setText(asmLine);
+            model->setItem(row, 0, asmItem);
+            row++;
+        }
+        ui->asmView->setModel(model);
+    }
+}
+
+/**
+ * Set current chosen function symbol for Disassembler
+ */
+void ResultsDisassemblyPage::setData(const Data::Symbol &symbol) {
+    m_curSymbol = symbol;
+
+    if (m_curSymbol.symbol.isEmpty()) {
+        return;
+    }
+    // If binary is not found at the specified path, use current binary file located at the application path
+    if (!QFile::exists(m_curSymbol.path)) {
+        m_curSymbol.path = m_appPath + QDir::separator() + m_curSymbol.binary;
+    }
+    // If binary is still not found, trying to find it in extraLibPaths
+    if (!QFile::exists(m_curSymbol.path)) {
+        QStringList dirs = m_extraLibPaths.split(QLatin1String(":"));
+        foreach (QString dir, dirs) {
+            QDirIterator it(dir, QDir::Dirs, QDirIterator::Subdirectories);
+
+            while (it.hasNext()) {
+                QString dirName = it.next();
+                QString fileName = dirName + QDir::separator() + m_curSymbol.binary;
+                if (QFile::exists(fileName)) {
+                    m_curSymbol.path = fileName;
+                    break;
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Set path to perf.data
+ */
+void ResultsDisassemblyPage::setData(const Data::DisassemblyResult &data) {
+    m_perfDataPath = data.perfDataPath;
+    m_appPath = data.appPath;
+    m_extraLibPaths = data.extraLibPaths;
+    m_arch = data.arch.trimmed().toLower();
+
+    m_objdump = m_arch.startsWith(QLatin1String("arm")) ? QLatin1String("arm-linux-gnueabi-objdump") : QLatin1String(
```

Comment:
we need validation via `QStandardPaths::findExecutable` if either of these is found. if not, this whole feature needs to be disabled with an error message saying that the process isn't found

also: QStringLiteral instead of QLatin1String here

Reply:
furthermore, we need to give the user the ability to configure the arch <-> objdump mapping somehow in the settings

Reply:
> we need validation via `QStandardPaths::findExecutable` if either of these is found. if not, this whole feature needs to be disabled with an error message saying that the process isn't found

How should I disable the whole feature when it was not found? 
Is it not enough instead to check if QProcess was started successfully and advise to install appropriate objdump if was not?

Reply:
ok yes that could work too, but make sure that the error message is descriptive enough in such a case

Reply:
oh and don't forget to add objdump as a runtime dependency in cmake, i.e. `find_package` + `set_package_properties` with `TYPE RUNTIME`

Reply:
can you please at least add a TODO for the configuration of the objdump path/name?